internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -92,64 +78,74 @@ func (r *BundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ c
 	}()
 	u.UpdateStatus(updater.EnsureObservedGeneration(bundle.Generation))
 
-	pod := &corev1.Pod{}
-	if op, err := r.ensureUnpackPod(ctx, bundle, pod); err != nil {
-		u.UpdateStatus(updater.SetBundleInfo(nil), updater.EnsureBundleDigest(""))
-		return ctrl.Result{}, updateStatusUnpackFailing(&u, fmt.Errorf("ensure unpack pod: %w", err))
-	} else if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated || pod.DeletionTimestamp != nil {
-		updateStatusUnpackPending(&u)
-		return ctrl.Result{}, nil
+	s, err := r.Sourcer.For(bundle.Spec.Source.Type)
+	if err != nil {
+		u.UpdateStatus(updater.SetBundleInfo(nil), updater.EnsureResolvedSource(nil))
+		return ctrl.Result{}, updateStatusUnpackFailing(&u, fmt.Errorf("get bundle sourcer: %w", err))
 	}
-
-	switch phase := pod.Status.Phase; phase {
-	case corev1.PodPending:
-		r.handlePendingPod(&u, pod)
+	res, err := s.SourceContent(ctx, bundle)
+	if err != nil {
+		u.UpdateStatus(updater.SetBundleInfo(nil), updater.EnsureResolvedSource(nil))
+		return ctrl.Result{}, updateStatusUnpackFailing(&u, fmt.Errorf("source bundle content: %w", err))
+	}
+	switch res.State {
+	case source.StatePending:
+		updateStatusUnpackPending(&u, res)
+		return ctrl.Result{}, nil
+	case source.StateUnpacking:
+		updateStatusUnpacking(&u)
 		return ctrl.Result{}, nil
-	case corev1.PodRunning:
-		r.handleRunningPod(&u)
+	case source.StateUnpacked:
```

Comment:
Nice - this seems much more streamlined. Any value in moving the logic for handling an unpacked source to it's own Bundle controller method?

Reply:
(we can mark this as resolved - this feels like it's in nit territory anyways)